sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/InterfaceCouldBeFunctionalCheck.kt

```diff
@@ -58,6 +58,8 @@ class InterfaceCouldBeFunctionalCheck : AbstractCheck() {
 }
 
 private fun hasExactlyOneFunctionAndNoProperties(klass: KtClass): Boolean {
+    if (klass.superTypeListEntries.isNotEmpty()) return false
+
     // Note: other possible declarations are KtClass (classes, interfaces) and
     //       KtObjectDeclaration (companion objects), but they are allowed inside function interfaces
     var functionCount = 0
```

Comment:
Optional: it's not related to the current change but, in my opinion, this function could be more readable and consistent with its name by changing the return to:
```
    return klass.declarations.filterIsInstance<KtNamedFunction>().size == 1 && klass.declarations.all { it !is KtProperty ```

Reply:
Good point!